src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -135,24 +135,23 @@ public void startTask(String tableName, String topic, int partition) {
     // the table should be present before opening a channel so lets do a table existence check here
     createTableIfNotExists(tableName);
 
-    createOrUpdatePartitionsToChannelCache(partitionChannelKey, tableName);
+    createStreamingChannelForTopicPartition(partitionChannelKey, tableName);
   }
 
-  /* We will always open a new channel. This will either update the TopicPartitionChannel's streamingChannel or will create a new instance of TopicPartitionChannel. */
-  private void createOrUpdatePartitionsToChannelCache(
+  /**
+   * Always opens a new channel and creates a new instance of TopicPartitionChannel.
+   *
+   * <p>This is essentially a blind write to partitionsToChannel. i.e we dont check if it is present
+   * or not.
```

Comment:
Can you elaborate why this is a blind write?

Reply:
Blind write because partitionsToChannel map is not checked before puting the key. 
May be it is not 100% appropriate since it is mostly related to database?